proto/substrait/type.proto

```diff
@@ -203,6 +203,9 @@ message Type {
 
   message Struct {
     repeated Type types = 1;
+    // Optionally names for the fields in this struct. If provided, must match the length of "types". Unlike other "names"
+    // fields, this does _not_ recursively include inner field names.
```

Comment:
```suggestion
    // (Optional) names for the fields in this struct, corresponding to "types" (co-indexed).
```
I feel like the comment about "..._not_ recursively include inner field names" is unnecessary. The constraint that length(names) == length(types) should imply whether recursive field names are required are not.